src/loaders/ObjectLoader.js

```diff
@@ -772,13 +772,14 @@ class ObjectLoader extends Loader {
 
 				if ( data.fog !== undefined ) {
 
-					if ( data.fog.type === 'Fog' ) {
+					if ( data.fog.type === 'RangeFog' || data.fog.type === 'Fog' ) {
 
-						object.fog = new Fog( data.fog.color, data.fog.near, data.fog.far );
+						object.fog = new RangeFog( data.fog.color, data.fog.near, data.fog.far );
 
-					} else if ( data.fog.type === 'FogExp2' ) {
+					} else if ( data.fog.type === 'DensityFog' || data.fog.type === 'FogExp2' ) {
 
-						object.fog = new FogExp2( data.fog.color, data.fog.density );
+						const squared = ( data.fog.squared == undefined ) || !! data.fog.squared;
```

Comment:
Did you mean `===` here?

Reply:
The issue here is that I don't know if `undefined` is really stored as `undefined` in JSON, and that both `undefined` and `null` are "falsey" values, whereas DensityFog defaults to `squared=true`, as requested by someone at some time (sorry). I would prefer to have `squared=false` as default, due to realism, and that would certainly facilitate less error-prone saving, loading, and dirtiness checks. I mean, as long as the to-be-deprecated `FogExp2` still results in construction with `squared=true`, are there any compelling reasons to keep that as the default?